src/components/ChartPanel/chartCalculations.test.js

```diff
@@ -0,0 +1,63 @@
+import { calculateRateChange, matchPairRates } from './chartCalculations';
+
+test('return 0 when we only have one item in the array', () => {
+	expect(calculateRateChange([{ rate: 1 }])).toBe(0);
+});
+test('calculate negative rate change', () => {
+	const newRate = { rate: 1 };
+	const oldRate = { rate: 2 };
+	const data = [newRate, oldRate];
+	expect(calculateRateChange(data)).toBe(-0.5);
+});
+test('calculate positive rate change', () => {
+	const newRate = { rate: 5 };
+	const oldRate = { rate: 1 };
+	const data = [newRate, oldRate];
+	expect(calculateRateChange(data)).toBe(4);
+});
+
+test('return pair rates regarding timestamp', () => {
+	const ETH_RATES = [
+		{ rate: 1, timestamp: 32 },
+		{ rate: 10, timestamp: 30 },
+		{ rate: 100, timestamp: 22 },
+		{ rate: 1000, timestamp: 20 },
+		{ rate: 100, timestamp: 10 },
+		{ rate: 10, timestamp: 0 },
+	];
+	const BTC_RATES = [
+		{ rate: 7000, timestamp: 200 },
+		{ rate: 9000, timestamp: 38 },
+		{ rate: 10000, timestamp: 35 },
+		{ rate: 8500, timestamp: 22 },
+		{ rate: 8000, timestamp: 12 },
+		{ rate: 7500, timestamp: 10 },
+		{ rate: 7000, timestamp: 6 },
+		{ rate: 6500, timestamp: 2 },
+		{ rate: 6000, timestamp: 1 },
+	];
+	const rates = matchPairRates(BTC_RATES, ETH_RATES);
+	expect(rates).toEqual([
+		{ rate: 7000, timestamp: 200 },
+		{ rate: 9000, timestamp: 38 },
+		{ rate: 10000, timestamp: 35 },
+		{ rate: 85, timestamp: 22 },
+		{ rate: 80, timestamp: 12 },
```

Comment:
It seems you are missing rates when the quote changes. E.g you should also have rate updates at timestamps `20`, `30` and `32`.

Reply:
and can you also make sure circle ci is invoking `npm test` - doesn't seem like it is right now

Reply:
Because I look for the base rates only (sBTC here).

Reply:
> Because I look for the base rates only (sBTC here). 

Is it relevant or not though I wonder? A quote change impacts the overall price although it suggests  a tick occurred at the base even though it didn't. That being said - these ticks are approximations of real world events, as they are notifications from oracles about on-chain prices, not actual ticks based on orders on an exchange, so I think it's better to include the union of both ticks against the USD, no?

Reply:
That's a good question. Couldn't really figure it out.

Happy to use the union of both ticks.

Reply:
It will increase the complexity dramatically though, that's the only bad part.

Reply:
You mean it'll add to rendering complexity right? The JS part should be trivial for the browser's CPU allocation. You could try the union and experiment to see what impacts there are to rendering?

Reply:
Yeah it will be fine as we're handling max 2,000 ticks.

Reply:
Done.